src/solvers/flattening/boolbv.cpp

```diff
@@ -463,6 +459,15 @@ literalt boolbvt::convert_rest(const exprt &expr)
     return convert_quantifier(expr);
   else if(expr.id()==ID_exists)
     return convert_quantifier(expr);
+  else if(expr.id()==ID_let)
+  {
+    bvt bv=convert_let(to_let_expr(expr));
+
+    if(bv.size()!=1)
+      throw "convert_let returned non-bool bitvector";
```

Comment:
This should be an `INVARIANT`. Or, really, why does it not return a `literalt` instead?!

Reply:
There are both cases, i.e., one where let is used in a boolean context and one where it produces bit-vectors. Clarified the error message.